src/ac_cottage.c

```diff
@@ -0,0 +1,996 @@
+        if (cottage->arg1 == Cottage_ACTION_DOOR_OPEN) {
+            Cottage_move_wait_pl_in_init(cottage);
+            return;
+        }
+
+        Cottage_move_wait_pl_out_init(cottage);
+        return;
+    }
+
+    if (mDemo_Check(mDemo_TYPE_SPEAK, &cottage->actor_class) != FALSE) {
+        return;
+    }
+
+    door_action = Cottage_check_door_action(cottage, play);
+    cottage_data = Cottage_data_get(cottage);
+    if (door_action != Cottage_ACTION_NONE) {
+        cottage->arg1 = door_action;
+    }
+
+    if (door_action == Cottage_ACTION_DOOR_OPEN) {
+        mDemo_Request(mDemo_TYPE_DOOR, &cottage->actor_class, cottage_data->demo_door_open_proc);
+        return;
+    }
+
+    if (door_action == Cottage_ACTION_OPEN_IN_INIT) {
+        Cottage_move_open_in_init(cottage);
+        return;
+    }
+
+    if (door_action == Cottage_ACTION_OPEN_PL_OUT_INIT) {
+        Cottage_move_open_pl_out_init(cottage);
+        return;
+    }
+
+    if (door_action == Cottage_ACTION_NPC_OUT_INIT) {
+        Cottage_move_open_npc_out_init(cottage);
+        return;
+    }
+
+    if (door_action == Cottage_ACTION_SPEAK_GO_OUT) {
+        mDemo_Request(mDemo_TYPE_SPEAK, &cottage->actor_class, &Cottage_demo_speak_go_out);
+    }
+}
+
+static void Cottage_move_wait_pl_in(STRUCTURE_ACTOR* cottage, GAME_PLAY* play);
+
+static void Cottage_move_wait_pl_in_init(STRUCTURE_ACTOR* cottage) {
+    cottage->action_proc = &Cottage_move_wait_pl_in;
+}
+
+static void Cottage_move_wait_pl_in(STRUCTURE_ACTOR* cottage, GAME_PLAY* play) {
+    if ((*GET_PLAYER_ACTOR_NOW()->get_door_label_proc)(gamePT) != cottage) {
+        return;
+    }
+
+    mBGMPsComp_make_ps_wipe(0x249);
+    Cottage_move_open_in_init(cottage);
+}
+
+static void Cottage_move_wait_pl_out(STRUCTURE_ACTOR* cottage, GAME_PLAY* play);
+
+static void Cottage_move_wait_pl_out_init(STRUCTURE_ACTOR* cottage) {
+    cottage->action_proc = &Cottage_move_wait_pl_out;
+}
+
+static void Cottage_move_wait_pl_out(STRUCTURE_ACTOR* cottage, GAME_PLAY* play) {
+    if (mPlib_check_player_outdoor_start(&play->game) == FALSE) {
+        return;
+    }
+
+    Cottage_move_open_pl_out_init(cottage);
+}
+
+static void Cottage_move_open_in(STRUCTURE_ACTOR* cottage, GAME_PLAY* play);
+
+static void Cottage_move_open_in_init(STRUCTURE_ACTOR* cottage) {
+    const COTTAGE_DATA* cottage_data;
+    cKF_SkeletonInfo_R_c* keyframe;
+
+    keyframe = &cottage->keyframe;
+    cottage_data = Cottage_data_get(cottage);
+
+    cKF_SkeletonInfo_R_init(keyframe, keyframe->skeleton, cottage_data->in_anim, 1.0f, 51.0f, 1.0f, 0.5f, 0.0f, cKF_FRAMECONTROL_STOP,
+                            NULL);
+    cottage->action_proc = &Cottage_move_open_in;
+}
+
+static void Cottage_move_next_init(STRUCTURE_ACTOR* cottage);
+
+static void Cottage_move_open_in(STRUCTURE_ACTOR* cottage, GAME_PLAY* play) {
+    const COTTAGE_DATA* cottage_data;
+
+    Cottage_set_doorSE(cottage);
+    if (cKF_SkeletonInfo_R_play(&cottage->keyframe) != TRUE) {
```

Comment:
cKF_SkeletonInfo_R_play returns an enum. This is `cKF_STATE_STOPPED`.

Additionally I'm not a fan of the returns when else could be used:

if (cKF_SKeletonInfo_R_play(&cottage->keyframe) == cKF_STATE_STOPPED) {
   if (cottage->arg1 == Cottage_ACTION_DOOR_OPEN) {
    // ..
   } else {
     Cottage_move_closed_init(cottage);
   }
}